### ISSUE TITLE
Maximum years supported in jalali is lower than 1416-10-30

### ISSUE THREAD

**Issue by amirhs712** [morilog/jalali#110]
Setting a date after `1416-10-30` would result in `1348-10-11` (1970).

### Steps to reproduce:
`\Morilog\Jalali\Jalalian::fromFormat('Y-m-d', '1416-12-01')`

Using version 3.1.2 of the package.

I need support for years bigger than 1416.

**Comment by morilog** [comment]
I wrote a test for it and it works properly
https://github.com/morilog/jalali/blob/7c3434548ecee790e894227e4717f895c0d3cd8d/tests/JalalianTest.php#L113

**Comment by amirhs712** [comment]
I ran your tests using `PHPUnit 4.8.36` and `PHP 7.2.7 (Zend engine 3.2.0)` which failed and behaved exactly as I described in the beginning.

But running the code using `PHP 7.4.3 (Zend engine 3.4.0)` was successful.
However I downloaded the latest `PHP 7.2.7 (2020 version)(Zend engine 3.2.0)` and ran the tests again with the same exact config and it passed.

I'm not sure what would be causing this issue.
